policy-controller/k8s/index/src/server.rs

```diff
@@ -0,0 +1,286 @@
+use crate::{authz::AuthzIndex, Index, Namespace, ServerRx, ServerSelector, ServerTx};
+use anyhow::{anyhow, bail, Result};
+use linkerd_policy_controller_core::{ClientAuthorization, InboundServer, ProxyProtocol};
+use linkerd_policy_controller_k8s_api::{self as k8s, policy, ResourceExt};
+use std::{
+    collections::{hash_map::Entry as HashEntry, BTreeMap, HashMap, HashSet},
+    sync::Arc,
+};
+use tokio::{sync::watch, time};
+use tracing::{debug, instrument, trace};
+
+#[derive(Debug, Default)]
+pub(crate) struct SrvIndex {
+    index: HashMap<String, Server>,
+}
+
+#[derive(Debug)]
+struct Server {
+    meta: ServerMeta,
+    authorizations: BTreeMap<String, ClientAuthorization>,
```

Comment:
assuming this is a btreemap because order matters? would be nice to have a comment about _why_ order matters...